### PR TITLE
End user/feature/search

### PR DESCRIPTION
**Pull request by NadaFeteiha**

### CODE REVIEW

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiEffect.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package presentation.search
+
+sealed class SearchUiEffect {
+    data object NavigateToMeal : SearchUiEffect()
+    data object NavigateToRestaurant : SearchUiEffect()
```
Comment (Saif-Alhaider) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiEffect.kt:5]: why data object and not data class or object ?

**Review by Saif-Alhaider — APPROVED** [review]

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiEffect.kt (hunk 1 above)
Comment (NadaFeteiha) [reply client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiEffect.kt:5]: @Saif-Alhaider not done yet

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiState.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    val name: String = ""
+)
+
+//Mapper
+
+fun Meal.toUiState() = MealUiState(
+    id = id,
+    name = name,
+    image = image,
+    restaurantName = restaurantId,
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiState.kt:34]:
```suggestion
    restaurantName = restaurantName,
```
image

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt [hunk 3]
```diff
@@ -1,15 +1,183 @@
...
     override fun Content() {
-        Text(text = Resources.strings.search)
+        initScreen(getScreenModel())
     }
 
-}
\ No newline at end of file
+    override fun onEffect(effect: SearchUiEffect, navigator: Navigator) {
+        when (effect) {
+            is SearchUiEffect.NavigateToMeal -> TODO()
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:51]:
```suggestion
            is SearchUiEffect.NavigateToMeal -> {// TODO: Navigate to meal details}
```

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt [hunk 4]
```diff
@@ -1,15 +1,183 @@
...
+            }
+        }
+    }
+
+
+    @Composable
+    private fun Restaurant(
+        modifier: Modifier = Modifier,
+        restaurant: RestaurantUiState,
+        onClick: (String) -> Unit
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:109]:
```suggestion
        restaurant: RestaurantUiState,
        onClick: (String) -> Unit,
        modifier: Modifier = Modifier,
```

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt [hunk 5]
```diff
@@ -1,15 +1,183 @@
...
+                contentDescription = restaurant.name,
+            )
+
+            Text(
+                text = restaurant.name,
+                style = Theme.typography.body,
+                color = Theme.colors.contentPrimary,
+                maxLines = 2,
+                textAlign = TextAlign.Center
+            )
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:129]: image
**we just need an overflow ellipse**

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt [hunk 6]
```diff
@@ -1,15 +1,183 @@
...
+        }
+    }
+
+
+    @Composable
+    private fun MealCard(
+        modifier: Modifier = Modifier,
+        meal: MealUiState,
+        onClick: (String) -> Unit
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:138]:
```suggestion
        meal: MealUiState,
        onClick: (String) -> Unit,
        modifier: Modifier = Modifier,
```

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt [hunk 7]
```diff
@@ -1,15 +1,183 @@
...
+            ) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Text(
+                        text = meal.name,
+                        style = Theme.typography.title,
+                        color = Theme.colors.contentPrimary,
+                        maxLines = 1
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:163]:
```suggestion
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
```

Comment (mosayed01) [reply client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:163]: same for other texts

**Review by mosayed01 — APPROVED** [review]
Great work! 👍❤️ However, I noticed some comments that need to be addressed.

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiState.kt (hunk 2 above)
Comment (NadaFeteiha) [reply client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchUiState.kt:34]: I depend on real data not fake and in the backend we need to return the restaurant name and it's not done yet

File: client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt (hunk 3 above)
Comment (NadaFeteiha) [reply client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt:51]: there is no meal details screen yet

**Review by ahmednasserzaza — APPROVED** [review]
Great Work 👏

**Review by HaidyAbuGom3a — APPROVED** [review]

**Pull request merged**